File: Python/program/Motorcycle.py
#Saya Nur Ainun 2202046 mengerjakan LATIHAN PRAKTIKUM 4 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. 

from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenisMotor = ""
    __kapasitasTangki = ""

    #constructor
    def __init__(self, platNomor, merk, tahunProduksi, warna, jenisMotor, kapasitasTangki):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki

    # Getter and Setter jenis motor
    def getJenisMotor(self):
        return self.__jenisMotor
    
    def setJenisMotor(self, jenisMotor):
        self.__jenisMotor = jenisMotor

    # Getter and Setter kapasitas tangki
    def getKapasitasTangki(self):
        return self.__kapasitasTangki
    
    def setKapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki

File: C++/program/Motorcycle.cpp
#pragma once

// memasukan liblary yang di gunakan termasuk kelas induk (kelas Vehicle)
//dari kelas car

#include <iostream>
#include <string>
#include "Vehicle.cpp"


// deklarasi namespace
using namespace std;

class Motorcyle : public Vehicle
{
private:
    string jenis_motor;
    string kapasitas_tangki;

public:

    //constructor : Overloading
    Motorcyle(string plat_nomor = "", string merk = "", string tahun_produksi = "", string warna = "", string jenis_motor = "" ,string kapasitas = "") 
    : Vehicle (plat_nomor, merk, tahun_produksi, warna)
    {
        this->jenis_motor = jenis_motor;
        this->kapasitas_tangki = kapasitas_tangki;
    } 

    void setjenis_motor(string jenis_motor)
    {
        this->jenis_motor = jenis_motor;
    }

    void setKapasitas_tangki(string kapasitas_tangki)
    {
        this->kapasitas_tangki = kapasitas_tangki;
    }

    string getjenis_motor()
    {
        return jenis_motor;
    }

    string getKapasitas_tangki()
    {
        return kapasitas_tangki;
    }

    void displayInfo_motor() {
        Vehicle::displayInfo();
        cout << "Jenis Motor: " << jenis_motor << "\nKapasitas Tangki: " << kapasitas_tangki;
    }

    //Destructor
    ~Motorcyle()
    {

    }

};